tsdb/head_wal.go

```diff
@@ -881,7 +886,6 @@ func (wp *wblSubsetProcessor) reuseBuf() []record.RefSample {
 
 // processWBLSamples adds the samples it receives to the head and passes
 // the buffer received to an output channel for reuse.
-// Samples before the minValidTime timestamp are discarded.
```

Comment:
This is not done (see [this comment](https://github.com/prometheus/prometheus/pull/12808/files#diff-e55e448a1ce649167194e1ced19ab2009f422ad678011e333414b8de8e62d311R893) down below).